Crashes on Loading Scenario without Error Message

### My Operating System:
Windows X
### Engine version:
1.9.0.0.7621
### Map name and version:
3rd Century BCE Wars
### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
Click the play button
### Do you have the exact error text? Please copy/paste if so
No text. Goes back to load screen.
### Instead of this error, what should have happened?
It should show the error log.
### Any additional information that may help:
[AnicentBCffa.zip](https://github.com/triplea-game/triplea/files/1566108/AnicentBCffa.zip)

Here's the exception that's being raised:

```
java.lang.reflect.InvocationTargetException
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1321)
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1296)
	at javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1348)
	at games.strategy.triplea.TripleA.startGame(TripleA.java:122)
	at games.strategy.engine.framework.startup.launcher.LocalLauncher.launchInNewThread(LocalLauncher.java:49)
	at games.strategy.engine.framework.startup.launcher.AbstractLauncher.lambda$0(AbstractLauncher.java:51)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.UnsupportedOperationException
	at java.util.AbstractList.remove(AbstractList.java:161)
	at games.strategy.triplea.ui.mapdata.MapData.getPlayerColor(MapData.java:444)
	at games.strategy.triplea.ui.screen.drawable.LandTerritoryDrawable.draw(LandTerritoryDrawable.java:29)
	at games.strategy.triplea.ui.screen.SmallMapImageManager.updateTerritoryOwner(SmallMapImageManager.java:79)
	at games.strategy.triplea.ui.MapPanel.initSmallMap(MapPanel.java:740)
	at games.strategy.triplea.ui.TripleAFrame.<init>(TripleAFrame.java:258)
	at games.strategy.triplea.TripleA.lambda$0(TripleA.java:124)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:353)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

The property `color.Scythians` is not defined, and the code attempts to pull off the next available default color.  However, because the default color list is immutable, this operation fails, and the exception is raised.

The fact that the default color list is immutable might be a regression.  I'll look into it further.

Note that after fixing the default colors issue, the following exception is now raised:

```
java.lang.reflect.InvocationTargetException
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1321)
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1296)
	at javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1348)
	at games.strategy.triplea.TripleA.startGame(TripleA.java:122)
	at games.strategy.engine.framework.startup.launcher.LocalLauncher.launchInNewThread(LocalLauncher.java:49)
	at games.strategy.engine.framework.startup.launcher.AbstractLauncher.lambda$0(AbstractLauncher.java:51)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Image Not Found:flags/GermanicTribes_small.png
	at games.strategy.triplea.image.ImageFactory.getImage(ImageFactory.java:35)
	at games.strategy.triplea.image.ImageFactory.getImage(ImageFactory.java:28)
	at games.strategy.triplea.image.FlagIconImageFactory.getSmallFlag(FlagIconImageFactory.java:26)
	at games.strategy.triplea.ui.CommentPanel.createComponents(CommentPanel.java:98)
	at games.strategy.triplea.ui.CommentPanel.init(CommentPanel.java:62)
	at games.strategy.triplea.ui.CommentPanel.<init>(CommentPanel.java:58)
	at games.strategy.triplea.ui.TripleAFrame.<init>(TripleAFrame.java:261)
	at games.strategy.triplea.TripleA.lambda$0(TripleA.java:124)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:353)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

I don't know if there's supposed to be support for providing a default small flag if the map doesn't provide one.  Otherwise, the 3rdCentury_BCE_Wars map will need to include one for the new GermanicTribes player (and presumably one for any other new player not present in 270BC).

--- a/src/main/java/games/strategy/debug/GenericConsole.java
+++ b/src/main/java/games/strategy/debug/GenericConsole.java
@@ -14,6 +14,7 @@ import javax.swing.SwingConstants;
 import javax.swing.SwingUtilities;
 import javax.swing.WindowConstants;
 
+import games.strategy.triplea.settings.ClientSetting;
 import games.strategy.ui.SwingAction;
 
 /**
@@ -63,7 +64,7 @@ public abstract class GenericConsole extends JFrame {
    */
   protected void displayStandardError() {
     final SynchedByteArrayOutputStream out = new SynchedByteArrayOutputStream(System.err);
-    final ThreadReader reader = new ThreadReader(out, textArea, true, getConsoleInstance());
+    final ThreadReader reader = new ThreadReader(out, textArea, Boolean.TRUE::booleanValue, getConsoleInstance());
     final Thread thread = new Thread(reader, "Console std err reader");
     thread.setDaemon(true);
     thread.start();
@@ -73,7 +74,8 @@ public abstract class GenericConsole extends JFrame {
 
   protected void displayStandardOutput() {
     final SynchedByteArrayOutputStream out = new SynchedByteArrayOutputStream(System.out);
-    final ThreadReader reader = new ThreadReader(out, textArea, false, getConsoleInstance());
+    final ThreadReader reader =
+        new ThreadReader(out, textArea, ClientSetting.SHOW_CONSOLE_ALWAYS::booleanValue, getConsoleInstance());
     final Thread thread = new Thread(reader, "Console std out reader");
     thread.setDaemon(true);
     thread.start();

--- a/src/main/java/games/strategy/debug/ThreadReader.java
+++ b/src/main/java/games/strategy/debug/ThreadReader.java
@@ -1,5 +1,7 @@
 package games.strategy.debug;
 
+import java.util.function.BooleanSupplier;
+
 import javax.swing.JTextArea;
 
 import games.strategy.util.ThreadUtil;
@@ -8,14 +10,17 @@ final class ThreadReader implements Runnable {
   private static final int CONSOLE_UPDATE_INTERVAL_MS = 100;
   private final JTextArea text;
   private final SynchedByteArrayOutputStream in;
-  private final boolean displayConsoleOnWrite;
+  private final BooleanSupplier displayConsoleOnWriteSupplier;
   private final GenericConsole parentConsole;
 
-  ThreadReader(final SynchedByteArrayOutputStream in, final JTextArea text, final boolean displayConsoleOnWrite,
+  ThreadReader(
+      final SynchedByteArrayOutputStream in,
+      final JTextArea text,
+      final BooleanSupplier displayConsoleOnWriteSupplier,
       final GenericConsole parentConsole) {
     this.in = in;
     this.text = text;
-    this.displayConsoleOnWrite = displayConsoleOnWrite;
+    this.displayConsoleOnWriteSupplier = displayConsoleOnWriteSupplier;
     this.parentConsole = parentConsole;
   }
 
@@ -23,7 +28,7 @@ final class ThreadReader implements Runnable {
   public void run() {
     while (true) {
       text.append(in.readFully());
-      if (displayConsoleOnWrite && !parentConsole.isVisible()) {
+      if (displayConsoleOnWriteSupplier.getAsBoolean() && !parentConsole.isVisible()) {
         parentConsole.setVisible(true);
       }
       if (!ThreadUtil.sleep(CONSOLE_UPDATE_INTERVAL_MS)) {

--- a/src/main/java/games/strategy/triplea/settings/ClientSetting.java
+++ b/src/main/java/games/strategy/triplea/settings/ClientSetting.java
@@ -92,6 +92,8 @@ public enum ClientSetting implements GameSetting {
 
   SHOW_BETA_FEATURES(false),
 
+  SHOW_CONSOLE_ALWAYS(false),
+
   TEST_LOBBY_HOST,
 
   TEST_LOBBY_PORT,

--- a/src/main/java/games/strategy/triplea/settings/ClientSettingSwingUiBinding.java
+++ b/src/main/java/games/strategy/triplea/settings/ClientSettingSwingUiBinding.java
@@ -115,6 +115,13 @@ enum ClientSettingSwingUiBinding implements GameSettingUiBinding<JComponent> {
           + "under development and potentially may not be working yet.\n"
           + "Restart to fully activate"),
 
+  SHOW_CONSOLE_ALWAYS(
+      "Show Console Always",
+      SettingType.TESTING,
+      ClientSetting.SHOW_CONSOLE_ALWAYS,
+      "Enable to show the console when any message is written to the console. "
+          + "Disable to show the console only when an error message is written to the console."),
+
   MAP_LIST_OVERRIDE_BINDING(
       "Map List Override",
       SettingType.TESTING,

